pkg/offsetguess/offsetguess.go

```diff
@@ -0,0 +1,293 @@
+// +build linux
+
+package offsetguess
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/iovisor/gobpf/elf"
+
+	"github.com/weaveworks/tcptracer-bpf/pkg/byteorder"
+)
+
+type tcpTracerState uint64
+
+const (
+	uninitialized tcpTracerState = iota
+	checking
+	checked
+	ready
+)
+
+type guessWhat uint64
+
+const (
+	guessSaddr guessWhat = iota
+	guessDaddr
+	guessFamily
+	guessSport
+	guessDport
+	guessNetns
+	guessDaddrIPv6
+)
+
+type tcpTracerStatus struct {
```

Comment:
The linked paragraph doesn't mention anything about padding or alignment of structs.

When using CGO, the Go compiler generates an intermediate type which is used to safely access a struct. In our case, we do not use CGO, therefore I am asking whether we can assume that underlying representation of the same struct in C and in Go is the same?